src/cmd-buildprep

```diff
@@ -61,11 +61,11 @@ def main():
         subprocess.check_call(['cp-reflink', BUILDFILES['list'], BUILDFILES['sourcedata']])
         builds = Builds()
 
-    if not builds or builds.is_empty():
-        print("Remote has no builds!")
+    if not builds or builds.is_empty_for_arch(basearch=args.arch):
+        print(f"Remote has no builds for {args.arch}!")
         return
 
-    buildid = args.build or builds.get_latest()
+    buildid = args.build or builds.get_latest_for_arch(basearch=args.arch)
```

Comment:
Hmm, I think here we want something like

```suggestion
    buildid = args.build or builds.get_latest()
    if args.arch not in builds.get_build_arches(buildid):
        print(f"No {args.arch} artifacts for build {buildid}")
        return
```

I think we probably always want to buildprep from the latest build ID. E.g. it's used by `cosa build` to determine the next build ID.

Reply:
That should address the problem I was hitting. I'll do that.